bigtable-hbase-1.x-parent/bigtable-hbase-1.x/src/test/java/com/google/cloud/bigtable/hbase1_x/TestMetrics.java

```diff
@@ -165,50 +160,89 @@ public void rowMutations() throws IOException {
 
     Stopwatch stopwatch = Stopwatch.createStarted();
     table.mutateRow(row);
-    long elapsed = stopwatch.elapsed(TimeUnit.MILLISECONDS);
+    long methodInvocationLatency = stopwatch.elapsed(TimeUnit.MILLISECONDS);
 
     fakeDataService.popLastRequest();
     long latency =
         fakeMetricRegistry
             .results
             .get("google-cloud-bigtable.grpc.method.MutateRow.rpc.latency")
             .get();
-    Assert.assertTrue(latency <= elapsed + 50 && latency >= elapsed - 50);
+    Truth.assertThat(latency)
+        .isIn(
+            Range.closed(FakeDataService.getMutateRowServerSideLatency(), methodInvocationLatency));
   }
 
   @Test
   public void appendFailure() throws IOException {
     Table table = connection.getTable(TABLE_NAME);
     Append append = new Append(rowKey);
     append.add(columnFamily, qualifier, value);
+    Stopwatch stopwatch = Stopwatch.createUnstarted();
     try {
+      stopwatch.start();
       table.append(append);
       Assert.fail("operation should have failed");
     } catch (Exception e) {
+      long methodInvocationLatency = stopwatch.elapsed(TimeUnit.MILLISECONDS);
       fakeDataService.popLastRequest();
       long failureCount =
           fakeMetricRegistry
               .results
               .get("google-cloud-bigtable.grpc.method.ReadModifyWriteRow.failure")
               .get();
+      long operationLatency =
+          fakeMetricRegistry
+              .results
+              .get("google-cloud-bigtable.grpc.method.ReadModifyWriteRow.operation.latency")
+              .get();
+
       Assert.assertEquals(1, failureCount);
+      Truth.assertThat(operationLatency)
+          .isIn(
+              Range.closed(
+                  FakeDataService.getReadModifyWriteRowServerSideLatency(),
+                  methodInvocationLatency));
     }
   }
 
   private static class FakeDataService extends BigtableGrpc.BigtableImplBase {
 
+    private Stopwatch readRowsStopwatch = Stopwatch.createUnstarted();
+    private Stopwatch readModifyWriteRowStopwatch = Stopwatch.createUnstarted();
+    private Stopwatch mutateRowStopwatch = Stopwatch.createUnstarted();
+
+    private static long readRowServerSideLatency;
+    private static long readModifyWriteRowServerSideLatency;
+    private static long mutateRowServerSideLatency;
+
     final ConcurrentLinkedQueue requests = new ConcurrentLinkedQueue();
 
     @SuppressWarnings("unchecked")
     <T> T popLastRequest() {
       return (T) requests.poll();
     }
 
+    public static long getReadRowServerSideLatency() {
+      return readRowServerSideLatency;
+    }
+
+    public static long getMutateRowServerSideLatency() {
+      return mutateRowServerSideLatency;
+    }
+
+    public static long getReadModifyWriteRowServerSideLatency() {
+      return readModifyWriteRowServerSideLatency;
+    }
+
     @Override
     public void readRows(
```

Comment:
I think this needs to be synchronized since you are updating latency from one thread and reading it from another